packages/outbox-core/lib/storage.ts

```diff
@@ -0,0 +1,32 @@
+import type { CommonEventDefinition } from '@message-queue-toolkit/schemas'
+import type { OutboxAccumulator } from './accumulators'
+import type { OutboxEntry } from './objects'
+
+/**
+ * Takes care of persisting and retrieving outbox entries.
+ *
+ * Implementation is required:
+ * - in order to fulfill at least once delivery guarantee, persisting entries should be performed inside isolated transaction
+ * - to return entries in the order they were created (UUID7 is used to create entries in OutboxEventEmitter)
+ * - returned entries should not include the ones with 'SUCCESS' status
+ */
+export interface OutboxStorage<SupportedEvents extends CommonEventDefinition[]> {
+  create(
```

Comment:
1) Maybe method name might be a bit more explicit, so that it's clearer what we are creating? E. g. `createEvent`

2) should it be possible to pass a transaction entity (e. g. Prisma TransactionClient) as a parameter here, so that the transactional guarantee is preserved?

Reply:
1. Sure, `createEvent` sounds good. On it.
2. I think it's up to the client to wrap this with transaction depending on the outbox adapter that will be provided.

For example, following code:
```js
contentLanguageRepository.create(contentLangauge)
eventEmitter.emit('content_langauge.added', payload);
```

It needs to be converted to:
```js
await prisma.$transaction(async () => {
  contentLanguageRepository.create(contentLangauge)
  outboxEmitter.emit('content_langauge.added', payload);
})
```

- Both rows are atomically added to the db.
- Code works with the assumption that `outboxStorage` configured for `outboxEmitter` uses prisma adapter with the same connection like for `contentLanguageRepository`.

Did you have different idea on how to address it?